persistence/src/main/java/cz/muni/fi/pa165/mushrooms/entity/Mushroom.java

```diff
@@ -0,0 +1,96 @@
+package cz.muni.fi.pa165.mushrooms.entity;
+
+import cz.muni.fi.pa165.mushrooms.enums.MushroomType;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.validation.constraints.NotNull;
+
+/**
+ *
+ * @author Lindar84
+ */
+@Entity
+public class Mushroom {
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    @Column(nullable=false, unique = true)
+    private String name;
+
+    @NotNull
+    @Column(nullable=false)
+    private MushroomType type;
+
+    @NotNull
+    @Column(nullable=false)
+    private String intervalOfOccurence;
+
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public MushroomType getType() {
+        return type;
+    }
+
+    public void setType(MushroomType type) {
+        this.type = type;
+    }
+
+    public String getIntervalOfOccurence() {
+        return intervalOfOccurence;
+    }
+
+    // "... in the following string format: "June - July" (month - month)"
+    // - is String enought? Wouldn't we need to use the dates for some sorting of mushrooms?
+    public void setIntervalOfOccurence(String startMonth, String endMonth) {
```

Comment:
If you want, we can use Dates as well (I would prefer it), but it would mean a bit more work. For now, let's leave it as strings.